### PR TITLE
Regions port: c2l_ord

### PR DESCRIPTION
**Pull request by oelbert**
Incorporating `region` syntax into c2l_ord and fusing the stencils into fewer calls.

### CODE REVIEW

**Comment by jdahm** [comment]
There is only 1 call to `compute_cubed_to_latlon`, in `fv_dynamics.py`. Could you move the stencil and driver method `compute_cubed_to_latlon` there?

File: fv3core/stencils/c2l_ord.py
Comment (jdahm) [inline fv3core/stencils/c2l_ord.py:88]: This function isn't needed any more. Can you inline the stencil call into the driver `compute_cubed_to_latlon`?

**Review by jdahm — CHANGES_REQUESTED** [review]
Can you add a docstring with in/out/inout specifications in the driver `compute_cubed_to_latlon`?

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
@rheacangeo It seems like this passes, but it keeps getting blocked by other Jenkins tests.

File: fv3core/stencils/c2l_ord.py
Comment (rheacangeo) [inline fv3core/stencils/c2l_ord.py:84]: 'at latitude-longitude coordinates' might make it more clear

File: fv3core/stencils/fv_dynamics.py
Comment (rheacangeo) [inline fv3core/stencils/fv_dynamics.py:211]: I would prefer the change from 'comm' to 'cube' be done to the whole stack in one separate PR, after discussing whether we want that change (sounded like you and Jeremy prefer it, I am less keen on it, but don't have a super strong objection).  But this is not a big deal, just introduces one more inconsistency in a sea of many. I think 'comm'  makes more sense because this model is not going to be anything but a cubed sphere, and cube would make more sense if we had multiple kinds of communicators present. You had suggested 'cube_comm', which is more clear it is a cubed sphere communicator, but that's a little long. I'm not going to block this PR on this point, just wanted to express my perspective. If 'cube' is popular I'm not going to stop it.

**Review by rheacangeo — COMMENTED** [review]
Looks good, just a couple of documentation related comments.

**Review by rheacangeo — APPROVED** [review]
Looks great! approved conditional on the tests passing!

**Review by jdahm — APPROVED** [review]

**Pull request merged**
